src/Silk.Core.Logic/Startup.cs

```diff
@@ -80,10 +97,75 @@ private static IHostBuilder ConfigureServices(IHostBuilder builder)
             {
                 var config = context.Configuration;
                 AddDatabases(services, config.GetConnectionString("core"));
+
                 services.AddScoped(typeof(ILogger<>), typeof(Shared.Types.Logger<>));
+
+                services.AddSingleton(new DiscordShardedClient(DiscordConfigurations.Discord));
+
+                services.AddMemoryCache(option => option.ExpirationScanFrequency = TimeSpan.FromSeconds(30));
+
+                services.AddHttpClient(StringConstants.HttpClientName, client => client.DefaultRequestHeaders.UserAgent.ParseAdd($"Silk Project by VelvetThePanda / v{StringConstants.Version}"));
+                services.AddSingleton(_ => new BotConfig(context.Configuration));
+
+
+                services.AddTransient<ConfigService>();
+                services.AddTransient<GuildContext>();
+                services.AddSingleton<AntiInviteCore>();
+                services.AddTransient<RoleAddedHandler>();
+                services.AddTransient<MemberAddedHandler>();
+                services.AddTransient<RoleRemovedHandler>();
+                services.AddSingleton<BotExceptionHandler>();
+                services.AddSingleton<SerilogLoggerFactory>();
+                services.AddTransient<MessageRemovedHandler>();
```

Comment:
Based off what I'm seeing here I suspect that you want a new handler produced every time that someone has a role removal or something. In that case, you might want to make getting the service lazy somehow. However, this is just a random thought. I have no idea if it's worth it.